modules/apps/object/object-test/src/testFunctional/tests/ObjectRootModel.testcase

```diff
@@ -0,0 +1,126 @@
+			}
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "Root Object");
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Object Root");
+
+			ObjectAdmin.goToFieldsTab();
+
+			ObjectAdmin.addObjectFieldViaUI(
+				fieldLabel = "Text Field",
+				fieldType = "Text");
+
+			for (var object1andObject2 : list "ObjectRoot-ObjectA,ObjectA-ObjectB,ObjectB-ObjectC") {
+				var objectName1 = StringUtil.extractFirst(${object1andObject2}, "-");
+				var objectName2 = StringUtil.extractLast(${object1andObject2}, "-");
+
+				ObjectAdmin.addObjectRelationshipViaAPI(
+					objectName_1 = ${objectName1},
+					objectName_2 = ${objectName2},
+					relationshipLabel = "Relationship ${object1andObject2}",
+					relationshipName = "relationship${objectName1}${objectName2}",
+					relationshipType = "oneToMany");
+			}
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectRootModel.bindObjects(
+				objectLabel = "Object C",
+				relationshipLabel = "Relationship ObjectB-ObjectC,Relationship ObjectA-ObjectB,Relationship ObjectRoot-ObjectA");
+		}
+
+		task ("And Given a grandchild object definition with a 1:M relationship with an unbound object definition") {
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "ObjectC",
+				objectName_2 = "ObjectD",
+				relationshipLabel = "Relationship ObjectC-ObjectD",
+				relationshipName = "relationshipObjectBObjectD",
+				relationshipType = "oneToMany");
+		}
+
+		task ("When the user selects the root object path") {
+			Click(
+				key_objectLabel = "Object D",
+				locator1 = "ObjectAdmin#OBJECT_KEBAB_MENU");
+
+			MenuItem.clickNoError(menuItem = "Bind");
+
+			Click(
+				key_value = "",
+				locator1 = "ObjectRootModel#SELECT_OBJECT_RELATIONSHIP");
+
+			Click(
+				key_type = "Relationship ObjectC-ObjectD",
+				locator1 = "ObjectAdmin#SELECT_RELATIONSHIPS_TYPE_OPTION");
+		}
+
+		task ("Then the all bound parents objects are displayed as read-only in the modal") {
+			for (var levelNames : list "Object B (Relationship ObjectB-ObjectC),Object A (Relationship ObjectA-ObjectB),Object Root (Relationship ObjectRoot-ObjectA)") {
+				AssertTextEquals(
+					key_value = ${levelNames},
+					locator1 = "ObjectRootModel#SELECT_OBJECT_RELATIONSHIP",
+					value1 = ${levelNames});
+			}
+		}
+
+		task ("And When the user binds them") {
+			Button.click(button = "Bind to Root");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "ObjectRoot");
+		}
+
+		task ("Then the root model is updated and the unbound object definition becomes the great-grandchild") {
+			var logContent = Log.getLogFileContent();
+			var inheritanceName = JSONObject.getObjectInheritanceLevel(objectName = "ObjectD");
+
+			if (contains(${logContent}, "/o/c/objectroots/objectds")) {
+				echo("Inheritance level: ${InheritanceName}");
+			}
```

Comment:
Why we checking the logs instead of using the endpoint to verify the great-grandchild like it was suggested on the ticket? Right now, there isn't a way for this step to fail since we're not making any assertions. All we're doing is echoing out some text if the log contains `/o/c/objectroots/objectds`